### PR TITLE
perf: optimize docker builds

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

closes: #1697

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: contrib/images/umee.e2e.dockerfile [hunk 1]
```diff
@@ -23,7 +26,6 @@ FROM ubuntu:rolling
 COPY --from=builder /go/pkg/mod/github.com/\!cosm\!wasm/wasmvm\@v*/internal/api/libwasmvm.*.so /usr/lib/
 COPY --from=builder /go/bin/* /usr/local/bin/
 COPY --from=builder /src/peggo-v*/peggo /usr/local/bin/
-RUN apt-get update && apt-get install ca-certificates -y
```
Comment (gsk967) [inline contrib/images/umee.e2e.dockerfile:26]: We need this one, price-feeder try to get the price from api request `api/v1/prices`

File: contrib/images/price-feeder.dockerfile [hunk 2]
```diff
@@ -14,6 +19,7 @@ RUN WASMVM_VERSION=$(go list -m github.com/CosmWasm/wasmvm | cut -d ' ' -f 2) &&
 # Build the binary
 RUN cd price-feeder && LEDGER_ENABLED=false BUILD_TAGS=muslc LINK_STATICALLY=true make install
 
+## Prepare the final clear binary
 FROM alpine:3.17
 RUN apk add bash curl jq
```
Comment (gsk967) [inline contrib/images/price-feeder.dockerfile:24]: We need to `ca-certificates` also for price-feeder
```
RUN apk update && apk add ca-certificates
```

**Review by gsk967 — COMMENTED** [review]
We need to add `ca-certificates` to docker final image to support `api requests` for price-feeder

**Comment by zarazan** [comment]
Could the wasm install be moved above the file copy as well?

File: contrib/images/umee.e2e.dockerfile (hunk 1 above)
Comment (robert-zaremba) [reply contrib/images/umee.e2e.dockerfile:26]: ah, ok, thank you!

File: contrib/images/price-feeder.dockerfile (hunk 2 above)
Comment (robert-zaremba) [reply contrib/images/price-feeder.dockerfile:24]: why do we need it in price feeder?

**Comment by robert-zaremba** [comment]
> Could the wasm install be moved above the file copy as well?

I don't think it will change anything: this is network query, so we will download it each time anyway.

File: contrib/images/price-feeder.dockerfile (hunk 2 above)
Comment (robert-zaremba) [reply contrib/images/price-feeder.dockerfile:24]: `apk update` - apk will run it with `add`

**Review by gsk967 — APPROVED** [review]
LGTM

**Pull request merged**